
syntax = "proto2";
package addressBook;

message Person{
    required string name = 1;
    required int32 id = 2;
    optional string email =3;

    enum PhoneType{
        MOBILE=0;
        HOME =1;
        WORK =2;
    }
    message PhoneNumber{
        required string number =1;
        optional PhoneType type = 2 [default = HOME];

    }
    repeated PhoneNumber phone = 4;
}
message AddressBook{
    repeated Person person =1;
}
message GetPersonRequest{
    required string name =1;
}
message GetPersonResponse{
    required Person requestedPerson =1;
}
message AddPersonResponse{
    required string msg =1;
}
message GetAllResponse{
    repeated Person person =1;
}
message GetAllRequest{
    optional string dummy =1;
}
service AddressBookService{
    rpc AddPerson(Person) returns (AddPersonResponse);
    rpc GetPerson(GetPersonRequest) returns(GetPersonResponse);
    rpc GetAll(GetAllRequest) returns (GetAllResponse);
}